src/plasma/plasma_store.cc

```diff
@@ -213,10 +238,9 @@ int create_object(Client *client_context,
   get_malloc_mapinfo(pointer, &fd, &map_size, &offset);
   assert(fd != -1);
 
-  entry = (object_table_entry *) malloc(sizeof(object_table_entry));
-  memset(entry, 0, sizeof(object_table_entry));
-  memcpy(&entry->object_id, &obj_id, sizeof(entry->object_id));
-  entry->info.obj_id = obj_id;
+  entry = new object_table_entry();
```

Comment:
Yes, we should really switch to STL containers sooner rather than later. This caused lots of headaches for me and will cause more for other people I feel. The problem here is that if we use malloc, C++ constructors don't get called, which is very bad if there are std::strings anywhere, as is the case for ObjectInfoT.